fair_mango/metrics/superset.py

```diff
@@ -0,0 +1,365 @@
+        ) = _initialise(data, sensitive, real_target, predicted_target, positive_target)
+        self.metrics = [SelectionRate, PerformanceMetric, ConfusionMatrix]
+
+    def evaluate(self) -> list[dict]:
+        """Calculate performance evaluation metrics for different subsets of
+        sensitive attributes. Ex:
+        [gender, race] → (gender), (race), (gender, race)
+
+        Returns
+        -------
+        list
+            A list of dictionaries, each containing the sensitive attributes
+            considered and their corresponding performance evaluation metric
+            results.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({
+        ...         'gender': ['male', 'male', 'male', 'male', 'female', 'female'],
+        ...         'race': ['white', 'white', 'black', 'black', 'white', 'white'],
+        ...         'real_churn': [1,0,1,0,0,1],
+        ...         'pred_churn': [0,0,1,0,0,1]
+        ... })
+        >>> result = super_set_performance_metrics(
+        ...     data=df,
+        ...     sensitive=['gender', 'race'],
+        ...     real_target=['real_churn'],
+        ...     predicted_target=['pred_churn'],
+        ... )
+        >>> result
+        [
+            {'sensitive': ('gender',),
+            'result': (['real_churn'],
+            [
+                {
+                    'sensitive': array(['male'], dtype=object),
+                    'selection_rate_in_data': array(0.5),
+                    'selection_rate_in_predictions': array(0.25),
+                    'accuracy': [0.75],
+                    'balanced accuracy': [0.75],
+                    'precision': [1.0],
+                    'recall': [0.5],
+                    'f1-score': [0.6666666666666666],
+                    'false_negative_rate': [0.5],
+                    'false_positive_rate': [0.0],
+                    'true_negative_rate': [1.0],
+                    'true_positive_rate': [0.5]
+                },
+                {
+                    'sensitive': array(['female'], dtype=object),
+                    ...
+                    'f1-score': [0.0],
+                    'false_negative_rate': [1.0],
+                    'false_positive_rate': [0.0],
+                    'true_negative_rate': [1.0],
+                    'true_positive_rate': [0.0]
+                }
+            ]
+            )
+            }
+        ]
+        """
+        results = []
+
+        for pair in self.pairs:
+            concatenated_results = SelectionRate(
+                data=self.data,
+                use_y_true=True,
+                sensitive=list(pair),
+                real_target=self.real_target,
+                predicted_target=self.predicted_target,
+                positive_target=self.positive_target,
+                label="selection_rate_in_data",
+            )()
+
+            for metric in self.metrics:
+                if metric is SelectionRate:
+                    result = SelectionRate(
```

Comment:
why is SelectionRate a performace metric here, but in `metrics.py` it is not in performance metric?

Reply:
It is not exactly a performance metric. It gives a result in a similar format to the performance metric class. But the way it is calculated is different. The performance metrics take as argument `y_true` and `y_pred` while the selection rate calculates the percentage of positive outcomes in the dataset (Using a simple `.mean()`).

Reply:
yes, but if it is a fairness metric then it does not have its place here

Reply:
It is not a fairness metric, technically we can consider it a performance metric actually but it has a totally different implementation compared to the other performance metrics. while the results are in the same format